plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -257,6 +259,28 @@ private static BackpressureRestHighLevelClient createClient(
     }
 
     private static AWSCredentialsProvider getAwsCredentialsProvider(AwsSecurityConfig config)
+    {
+        AWSCredentialsProvider iamUserCredentialsProvider = getIamUserCredentialsProvider(config);
+        return config.getIamRole()
+                .map(roleArn -> getIamRoleCredentialsProvider(roleArn, config.getRegion(), iamUserCredentialsProvider, config.getExternalId()))
+                .orElse(iamUserCredentialsProvider);
+    }
+
+    private static AWSCredentialsProvider getIamRoleCredentialsProvider(String roleArn,
+            String awsRegion,
+            AWSCredentialsProvider iamUserCredentialsProvider,
+            Optional<String> externalId)
+    {
+        STSAssumeRoleSessionCredentialsProvider.Builder credentialsProviderBuilder = new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, "trino-session")
+                .withStsClient(AWSSecurityTokenServiceClientBuilder.standard()
+                        .withRegion(awsRegion)
+                        .withCredentials(iamUserCredentialsProvider)
+                        .build());
+        externalId.ifPresent(credentialsProviderBuilder::withExternalId);
+        return credentialsProviderBuilder.build();
+    }
+
+    private static AWSCredentialsProvider getIamUserCredentialsProvider(AwsSecurityConfig config)
```

Comment:
This is misnamed. It might not be IAM user credentials (they could be an instance role or something of that sort).

In fact, I would just put this all together in a single method. There's little benefit in splitting it up into separate method (it doesn't improve clarity enough):

```java
AWSCredentialsProvider credentialsProvider = DefaultAWSCredentialsProviderChain.getInstance();

if (config.getAccessKey().isPresent() && config.getSecretKey().isPresent()) {
    credentialsProvider = new AWSStaticCredentialsProvider(new BasicAWSCredentials(
            config.getAccessKey().get(),
            config.getSecretKey().get()));
}

if (config.getIamRole().isPresent()) {
    STSAssumeRoleSessionCredentialsProvider.Builder credentialsProviderBuilder = new STSAssumeRoleSessionCredentialsProvider.Builder(config.getIamRole().get(), "trino-session")
            .withStsClient(AWSSecurityTokenServiceClientBuilder.standard()
                    .withRegion(config.getRegion())
                    .withCredentials(credentialsProvider)
                    .build());
    config.getExternalId().ifPresent(credentialsProviderBuilder::withExternalId);
    credentialsProvider = credentialsProviderBuilder.build();
}

return credentialsProvider;
```